### ISSUE TITLE
http4s Tutorial

### ISSUE THREAD

**Issue by noelwelsh** [http4s/http4s#5539]
As part of the documentation rework I want to write a tutorial on basic use of http4s. I'm planning to use [ScalaBridge](https://github.com/scalabridgelondon/lesson-plans/blob/master/web/0-intro.md) material as the basis for this (and use this in ScalaBridge 🐦 🐦 🪨 ). The tutorial will describe building the [Cheep](https://github.com/scalabridgelondon/cheep) application, which is perhaps the simplest web application one can build. This will cover routes, handling requests and sending results, and testing.

Tasks are:

* Port Cheep to new versions of stuff
* Write the text!

I'm planning to use Scala 3.1.0 and http4s 0.23.

**Comment by mfirry** [comment]
Hi @noelwelsh. I've given this a [try](https://github.com/mfirry/cheep/tree/3.1).
Spoiler: it does not compile _yet_. 😄 

Let me know if it's any useful.
Here's few things I had to "remove" (not because the _need_ to removed but just 'cause I couldn't make them work):
- all `scalacOptions` (though should be easy to fix/port)
- everything related to `scalafix`
- few `scalafmt` things

I think the first thing that needs fixing is removing `Blocker`.

**Comment by noelwelsh** [comment]
Looks good! I was planning on replacing Slinky with scalajs-react, as Slinky is not yet on Scala 3.

**Comment by mfirry** [comment]
Good point I missed: I'm not a big scalajs user. So that bit I'll not to touch too much.

**Comment by noelwelsh** [comment]
No worries. I'm happy to do that.

**Comment by mfirry** [comment]
Good. I'll try and work my way out of this `Blocker` thing (pun intended) and keep you posted here.

**Comment by mfirry** [comment]
Have another look. https://github.com/mfirry/cheep/tree/3.1
`backend` compiles. Let me know how you wanna move this further (and if). 
Cheers.

**Comment by armanbilge** [comment]
> Looks good! I was planning on replacing Slinky with scalajs-react, as Slinky is not yet on Scala 3.

Actually, slinky _is_ as of a couple of days ago. I assume a proper release is imminent, but until then there is a snapshot available: `0.6.8+6-998383e1`

- https://github.com/shadaj/slinky/pull/494

**Comment by noelwelsh** [comment]
> Have another look. https://github.com/mfirry/cheep/tree/3.1 `backend` compiles. Let me know how you wanna move this further (and if). Cheers.

Sorry about the delay. I've been heads down on some work stuff. This looks good, and the code is simpler than before. If it all compiles it should be good to merge. I'll take a look at the front end soon.

**Comment by noelwelsh** [comment]
> > Looks good! I was planning on replacing Slinky with scalajs-react, as Slinky is not yet on Scala 3.
> 
> Actually, slinky _is_ as of a couple of days ago. I assume a proper release is imminent, but until then there is a snapshot available: `0.6.8+6-998383e1`
> 
>     * [Scala 3 support shadaj/slinky#494](https://github.com/shadaj/slinky/pull/494)

Interesting! I might still look at scalajs-react because I think it's more inline with the Typelevel ecosystem, but Slinky is definitely a great project and a bit more approachable than scalajs-react.

**Comment by noelwelsh** [comment]
With the help of @mfirry Cheep has been ported to Scala 3, http4s 0.23.6, and scalajs-react 2. That's the programming side done. Next up is writing the text.

**Comment by armanbilge** [comment]
@noelwelsh would you consider using http4s-dom for the frontend instead of sttp? Happy to PR :)
https://github.com/http4s/http4s-dom

**Comment by noelwelsh** [comment]
Sure. I used sttp for two reasons:

* I needed to work with `IO` instead of `Future`
* I knew about it.

**Comment by mfirry** [comment]
hey @noelwelsh, was it really? I'm not sure I even made a PR to be honest. anyways, happy to have helped!

**Comment by mfirry** [comment]
Ah ok. Forgot you were able to do so on my fork. Totally fine! I'll see if I can help more.

**Comment by armanbilge** [comment]
@noelwelsh done :) https://github.com/scalabridgelondon/cheep/pull/4
